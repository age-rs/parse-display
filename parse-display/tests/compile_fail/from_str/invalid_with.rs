use parse_display::FromStr;

#[derive(FromStr, Debug, PartialEq)]
struct X {
    #[from_str(with = "not impl FromStrFormat")]
    x: u8,
}

fn main() {}
