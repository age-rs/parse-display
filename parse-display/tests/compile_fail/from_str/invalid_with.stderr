error[E0277]: the trait bound `&'static str: FromStrFormat<u8>` is not satisfied
 --> tests/compile_fail/from_str/invalid_with.rs:5:23
  |
5 |     #[from_str(with = "not impl FromStrFormat")]
  |                       ^^^^^^^^^^^^^^^^^^^^^^^^ the trait `FromStrFormat<u8>` is not implemented for `&'static str`
  |
  = note: required for the cast from `&&'static str` to `&dyn FromStrFormat<u8, Err = _>`
